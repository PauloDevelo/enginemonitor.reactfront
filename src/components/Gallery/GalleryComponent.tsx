import React from 'react';
import { Label, Button } from 'reactstrap';

import Image from './Image';

import { ImageModel } from '../../types/Types';
import FileChooserButton from './FileChooserButton';

import { faCamera, faPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import "./Gallery.css";

type Props = {
    images: ImageModel[],
    onClickThumbnail: (index: number) => void,
    onSelectFile: (file: File) => void,
    turnOnCamera: () => void
};

function GalleryComponent({ images, onClickThumbnail, onSelectFile, turnOnCamera }: Props){
    const thumbnails = images.map((image, index) => {
        return <Image key={image._uiId} image={image} onClickImage={() => onClickThumbnail(index)} />;
    });

	return(
        <div>
            <Label className="font-weight-bold">Gallery image</Label>
            <div className="p-1 border border-secondary rounded shadow gallery">
                <FileChooserButton onFileSelect={onSelectFile} className="float-right"/>
                <Button color="light" size="lg" onClick={turnOnCamera} className="float-right">
                    <span className="fa-layers fa-fw">
                        <FontAwesomeIcon icon={faCamera} size="lg"/>
                        <FontAwesomeIcon icon={faPlus} size="xs" transform="down-13 left-16"/>
                    </span>
                </Button>
                {thumbnails}
            </div>
        </div>
	);
}

export default React.memo(GalleryComponent);