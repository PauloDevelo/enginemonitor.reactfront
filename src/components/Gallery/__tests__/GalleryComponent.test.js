import React from 'react';
import { IntlProvider } from 'react-intl';
import { mount } from 'enzyme';
// eslint-disable-next-line no-unused-vars
import localforage from 'localforage';
import storageService from '../../../services/StorageService';
import ignoredMessages from '../../../testHelpers/MockConsole';

import GalleryComponent from '../GalleryComponent';

jest.mock('../../../services/ImageProxy');
jest.mock('localforage');
jest.mock('../../../services/StorageService');

describe('Component GalleryComponent', () => {
  const parentUiId = 'equipment_01';

  const images = [
    {
      name: 'DJI_1645.JPG',
      parentUiId,
      sizeInByte: 262985,
      thumbnailUrl: 'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg',
      url: 'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339188DJI_1645.JPG.jpeg',
      _uiId: 'dfc66510-f5f3-11e9-b3b1-f7086d79fbca',
    },
    {
      name: 'DSC_0016.JPG',
      parentUiId,
      sizeInByte: 325323,
      thumbnailUrl: 'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg',
      url: 'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358061DSC_0016.JPG.jpeg',
      _uiId: 'eb074110-f5f3-11e9-b3b1-f7086d79fbca',
    },
    {
      name: 'DSC_0005.JPG',
      parentUiId,
      sizeInByte: 239719,
      thumbnailUrl: 'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg',
      url: 'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420020DSC_0005.JPG.jpeg',
      _uiId: '0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca',
    },
  ];

  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('[@formatjs/intl Error MISSING_TRANSLATION]');
    ignoredMessages.push('a test was not wrapped in act');
  });

  beforeEach(() => {
    storageService.getUserStorage = jest.fn(() => ({
      getItem: jest.fn(() => Promise.resolve('data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCABCAGQDASIAAhEBAxEB/8QAHAAAAgMBAQEBAAAAAAAAAAAAAAUDBgcEAggB/8QANxAAAgEDAwIEAwYGAQUAAAAAAQIDAAQRBRIhMUEGEyJRYYGRByMyQnHBFBUWQ6Gx8WJyksLR/8QAGQEAAwEBAQAAAAAAAAAAAAAAAQIDBAAF/8QAIREAAgIBBAMBAQAAAAAAAAAAAAECERIDISJREzFhMkH/2gAMAwEAAhEDEQA/AM5Wza2laOf0OvUEcg0MWRfu4w59t1AvBJGxvLh3mAVISoB4B5Dd+/Wv0DLHAOQeeOtebNUxWmj8gimu51IhCy8IqL+bPYe5q1eBo7GLxXGmoWsUnpcIlyxUA46YwctjpnA+eKi8H6VpGv6gNMvLm4tbuQnyXSMOjHHAPcc85/11rYY/DlhFpf8AF6vFbT6xbQeUZ1PUqAAfieOtadPTy5r+C3boXaJNa6LDb2lvZoUhcmNnALoSTkg/OtOwHXDAEH3rJFbEqNj8wrUZb1YIFcqT8PlV4NstOKjsSy2sUq4ZAD7gUo1W0MFlK1udrgjDHtyKbtdRrkk5A5JFcOov5lrNjOAR0OO6nrVU5IjJRZTBcSQTTyJGEmmI8xgOWI4Ga7rCe52SmcDdFhw3uOv7UtvHIvie+VP4gewptbOrXUkLf3YwAf0FWi8a+mfG7+CvBi1or+VZmAHtk00vQBHsIJLdKVaowtNbdmQl/TISq5wDzz9aeX8PphdWGCDkj2rpSWFBUXlZWrsXvmr5TOF2g4UgDJ57j40UxGoxWmYzEr59WSp7/pRWWy+R88wacxzIrA45buT+tM7K9InRpbeMhTkpyAR3Bxz86VJfpDGk1xGqQzcKoJyffBB4PTmore+knW5SBXJJJiLtlgM9+OcDFYVGT3K+OTN78EeE7PTPENtrNnNbTW15bFo45TmWEkAnaeh7j3xmrjq0sS6PfZzvIbGVIzzjgnrXz54G+0Wfw7qcTXkRuLTJXaSQBnjKex+HxNalF43i1TTNRHrZJXUxbMnyxkE57CtiajGkJhjJM58xxuocgEEEksMCrBe67LIhjgHmKq4dwMEk8ZHbv/xVZKTTOAkCyLjLbjjP7mkt3c+K7aVbm3u7K32tsRDxwOhIYfvWHR8ztSZocVJ3Jmt2d/He6ebiDcSi4YOMc4rwbx5tNnZ+GIyQAGFUPRfEmtQabOuoS21zKxcIkABY/FipA+A4/arNo9xfTaLdre25jcRthgPxcZGAa9FTvZmVxxdCq9kR9RMQyG2jnaAOBTCaPcyDPJDDI/XFV/UZHTxBGqhskgcp/wBRHantxIQ6YY+aSCI/L68Hnjn/AJpp6kaSsWEeTF41KPU7wXTL5TxwmGUFwQ23gHHHf9qc3GpwS6XbRIxeZQqkKueQPeqbqWr6foBuEW59LOWKgAbm75J7DngA49qp2r+O5bhFW0BMTHBWNtgIB6Ej1H/A+FRlqW+JeOk2tzSZ9dgtZTFNq1naOv8Aaba5H6k96Kw59T1iSR3huJ44yxKrCNij5Cik59lPAuxXqO+cwXK3ME9uGAVE4K9+QeajikksbmXy5I1kIZQpGcA9vpTGVbG2t7e3/l8qSO4Ks0nQ/EYOaie0gkvJ7hxn7zAwfbHArlVfDoJkUFsqPc28mXZkLoiDIyDj5YBNPtH1rTdILiawuS5G3IORx+tQafbxm/N0ZdoaEgKBzktmvQ02S4u3H8V5YUIEjBOcYGaVzSdDRSk6Y+j8X6ZKfu47m2UEbpGxgfPn/VWax8RaPCqNLrVo6lfUhuDntx/v61n9s9sqCC8WR4n5Ei43L09+orw1n4YM4dtRn8zPeFuv0xTLdAkqexoNv4+0eWZo3inkG4KrxMWHT44ple+LLSS8imYXCwLEEaNj1wPgfjVCWKKGUpbyN5QXe00jL9Ao5rosJZNWZQHZRhmye+CBRSxTbGwUmki7W32heHrV3kFrceZtKbiobgjBPJpDq/j+9nnxp9vJsUbdwG0kexI5+hFIrnTr+3a9B8loFjOXB3fmHAPvgH5U/wDDsUMVpq2RGsi3LCNmAzkjKgfWkVNX7BjG7RVbthqqyTajDILzI2GFgN3BHOc9to+X61Fb2Vpa2OZIJpLhiUKoMkEcHGAe9WDTdVmm1Ii9IliVN+No6gjmnniuyg0nV72CyQxgs7jBzgk84rn0VUkndFOh0Ca5TzI9OvmTOAWbn/VFSeuP8UzsW5yxJNFIU8i6RXtWeW+1+Njt3xRBuRgHjOKitLcDSY2eby380bo2RufUO+MdKsqeHWurkXbGaNiMYzjAHY1MtkTCLewAlmBxtPUHPPH1qsUlFIzN8m0VLTw6XhU70DPwOhrohuY4tculZ1QMRsYngHb79qsGkeF9UudSB1awuTH6hv3jGO3FXWDwJpxbd98p9sj/AOUcU3YqdbmVpIBKqmaOXaMejkCpIbKBormSVlYLjaAenIz29s1qz/ZvpkzCRpp8qOASDn/FUy5/ky6vHCGn2NwV8kBlIA4HHOcUPyylpr6I5mRN7ieFlAwETcSef+0U60y/hs9O0Vp5hAi+eHJB5BYEdAa0y28KWN3Zw3AMirKgfDAZGfepR4KsN4IAxjGAootWgKdGY21/cT3TR210slssUj4/CDgFmOODnHTPtUH9VXNhqeqiLyis05blM9Bjj6Vpn9AaULkzOZ2U9UD4FONP8B+FElEkcDh8YKTsGU/IihGGJPZejAhqMttKJo152leRnINXX7Sr64g1Kxu4eVuYt5AXOcqp/wDatm/prT7dFWDS7PaO6xqD/kfvXJcaJCuoyXbXDxh1CiOZMxrjjgngV0ru6GT+nzu+qCTaRA3AxylFfRX8pZuVKEfBBRSbdD2ZfMxGxQSB7UWxKXAKkqWxkjjNFFNERlrUCpQSBwSOKKKqITKzccnp71nl969G1ud/VMlxhJDyy+rse1FFBhL3pM0p0y1zI5+5T8x9hTVXbj1H60UVyCdIJKda8flP60UUwjGuiyOzsrOxUDgE8U5PIOfaiiiA43t4Q5+5j/8AEUUUVIc//9k=')),
      keys: jest.fn(() => Promise.resolve([
        'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg',
        'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg',
        'http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg',
      ])),
    }));
  });

  afterEach(() => {
    storageService.getUserStorage.mockRestore();
  });

  it('should render a spinner the time to get the answer from the back end.', async () => {
    // Arrange
    const onClickThumbnail = jest.fn();
    const addImage = jest.fn();

    // Act
    const galleryComponent = mount(
      <IntlProvider locale="en-US" timeZone="Asia/Kuala_Lumpur">
        <GalleryComponent parentUiId={parentUiId} isLoading images={[]} onClickThumbnail={onClickThumbnail} addImage={addImage} />
      </IntlProvider>,
    );

    // Assert
    const loading = galleryComponent.find('Loading');
    expect(loading.length).toBe(1);

    expect(onClickThumbnail).toHaveBeenCalledTimes(0);
    expect(addImage).toHaveBeenCalledTimes(0);

    expect(galleryComponent).toMatchSnapshot();
  });

  it('should render a list of image using the thumbnail url', async () => {
    // Arrange
    const onClickThumbnail = jest.fn();
    const addImage = jest.fn();

    // Act
    const galleryComponent = mount(
      <IntlProvider locale="en-US" timeZone="Asia/Kuala_Lumpur">
        <GalleryComponent parentUiId={parentUiId} images={images} onClickThumbnail={onClickThumbnail} addImage={addImage} />
      </IntlProvider>,
    );

    // Assert
    const thumbnails = galleryComponent.find('Memo(Thumbnail)');
    expect(thumbnails.length).toBe(3);
    for (let i = 0; i < 3; i++) {
      expect(thumbnails.at(i).props().image).toBe(images[i]);
    }

    expect(onClickThumbnail).toHaveBeenCalledTimes(0);
    expect(addImage).toHaveBeenCalledTimes(0);

    expect(galleryComponent).toMatchSnapshot();
  });

  it('should call onClickThumbnail function with the correct index when we click on a thumbnail', async () => {
    // Arrange
    const onClickThumbnail = jest.fn();
    const addImage = jest.fn();

    const galleryComponent = mount(
      <IntlProvider locale="en-US" timeZone="Asia/Kuala_Lumpur">
        <GalleryComponent parentUiId={parentUiId} images={images} onClickThumbnail={onClickThumbnail} addImage={addImage} />
      </IntlProvider>,
    );

    const thumbnails = galleryComponent.find('Memo(Thumbnail)');

    for (let index = 0; index < 3; index++) {
      // Act
      thumbnails.at(index).find('Loading').simulate('click');

      // Assert
      expect(onClickThumbnail.mock.calls[index][0]).toBe(index);
    }

    expect(onClickThumbnail).toHaveBeenCalledTimes(3);
    expect(addImage).toHaveBeenCalledTimes(0);
  });
});
