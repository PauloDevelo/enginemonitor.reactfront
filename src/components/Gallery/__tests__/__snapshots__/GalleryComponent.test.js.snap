// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component GalleryComponent should render a list of image using the thumbnail url 1`] = `
<Memo(GalleryComponent)
  addImage={[MockFunction]}
  images={
    Array [
      Object {
        "_uiId": "dfc66510-f5f3-11e9-b3b1-f7086d79fbca",
        "name": "DJI_1645.JPG",
        "parentUiId": "equipment_01",
        "sizeInByte": 262985,
        "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg",
        "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339188DJI_1645.JPG.jpeg",
      },
      Object {
        "_uiId": "eb074110-f5f3-11e9-b3b1-f7086d79fbca",
        "name": "DSC_0016.JPG",
        "parentUiId": "equipment_01",
        "sizeInByte": 325323,
        "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg",
        "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358061DSC_0016.JPG.jpeg",
      },
      Object {
        "_uiId": "0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca",
        "name": "DSC_0005.JPG",
        "parentUiId": "equipment_01",
        "sizeInByte": 239719,
        "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg",
        "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420020DSC_0005.JPG.jpeg",
      },
    ]
  }
  onClickThumbnail={[MockFunction]}
  parentUiId="equipment_01"
>
  <div
    className="flex-row top-padding-4px"
  >
    <Label
      className="font-weight-bold"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <label
        className="font-weight-bold"
      >
        <FormattedMessage
          defaultMessage="Gallery"
          description="Gallery title"
          id="gallerytitle"
          values={Object {}}
        >
          Gallery
        </FormattedMessage>
      </label>
    </Label>
    <div>
      <Memo(AddImageFileButton)
        addImage={[MockFunction]}
        className="float-right"
        parentUiId="equipment_01"
      >
        <Button
          aria-label="Edit"
          className="float-right"
          color="light"
          onClick={[Function]}
          tag="button"
        >
          <button
            aria-label="Edit"
            className="float-right btn btn-light"
            onClick={[Function]}
            type="button"
          >
            <span
              className="fa-layers fa-fw"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f03e",
                      "M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z",
                    ],
                    "iconName": "image",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="lg"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-image fa-w-16 fa-lg "
                  data-icon="image"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f067",
                      "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                    ],
                    "iconName": "plus",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="xs"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform="down-13 left-16"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 fa-xs "
                  data-icon="plus"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "transformOrigin": "-0.5625em 1.3125em",
                    }
                  }
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    style={Object {}}
                    transform="translate(224 256)"
                  >
                    <g
                      style={Object {}}
                      transform="translate(-512, 416)  scale(1, 1)  rotate(0 0 0)"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                        style={Object {}}
                        transform="translate(-224 -256)"
                      />
                    </g>
                  </g>
                </svg>
              </FontAwesomeIcon>
            </span>
          </button>
        </Button>
      </Memo(AddImageFileButton)>
    </div>
  </div>
  <div
    className="p-1 border border-secondary rounded shadow gallery"
  >
    <Memo(Thumbnail)
      image={
        Object {
          "_uiId": "dfc66510-f5f3-11e9-b3b1-f7086d79fbca",
          "name": "DJI_1645.JPG",
          "parentUiId": "equipment_01",
          "sizeInByte": 262985,
          "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg",
          "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339188DJI_1645.JPG.jpeg",
        }
      }
      key="dfc66510-f5f3-11e9-b3b1-f7086d79fbca"
      onClickImage={[Function]}
    >
      <Img
        alt="undefined - undefined"
        className="thumbnail grow"
        loader={
          <span
            className="spinnerContainer"
          >
            <Loading
              onClick={[Function]}
            />
          </span>
        }
        onClick={[Function]}
        src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg"
        storage={
          Object {
            "getItem": [MockFunction],
            "keys": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Promise {},
                },
              ],
            },
          }
        }
      >
        <span
          className="spinnerContainer"
        >
          <Loading
            onClick={[Function]}
          >
            <div
              className="lds-spinner"
              onClick={[Function]}
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </Loading>
        </span>
      </Img>
    </Memo(Thumbnail)>
    <Memo(Thumbnail)
      image={
        Object {
          "_uiId": "eb074110-f5f3-11e9-b3b1-f7086d79fbca",
          "name": "DSC_0016.JPG",
          "parentUiId": "equipment_01",
          "sizeInByte": 325323,
          "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg",
          "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358061DSC_0016.JPG.jpeg",
        }
      }
      key="eb074110-f5f3-11e9-b3b1-f7086d79fbca"
      onClickImage={[Function]}
    >
      <Img
        alt="undefined - undefined"
        className="thumbnail grow"
        loader={
          <span
            className="spinnerContainer"
          >
            <Loading
              onClick={[Function]}
            />
          </span>
        }
        onClick={[Function]}
        src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg"
        storage={
          Object {
            "getItem": [MockFunction],
            "keys": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Promise {},
                },
              ],
            },
          }
        }
      >
        <span
          className="spinnerContainer"
        >
          <Loading
            onClick={[Function]}
          >
            <div
              className="lds-spinner"
              onClick={[Function]}
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </Loading>
        </span>
      </Img>
    </Memo(Thumbnail)>
    <Memo(Thumbnail)
      image={
        Object {
          "_uiId": "0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca",
          "name": "DSC_0005.JPG",
          "parentUiId": "equipment_01",
          "sizeInByte": 239719,
          "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg",
          "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420020DSC_0005.JPG.jpeg",
        }
      }
      key="0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca"
      onClickImage={[Function]}
    >
      <Img
        alt="undefined - undefined"
        className="thumbnail grow"
        loader={
          <span
            className="spinnerContainer"
          >
            <Loading
              onClick={[Function]}
            />
          </span>
        }
        onClick={[Function]}
        src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg"
        storage={
          Object {
            "getItem": [MockFunction],
            "keys": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Promise {},
                },
              ],
            },
          }
        }
      >
        <span
          className="spinnerContainer"
        >
          <Loading
            onClick={[Function]}
          >
            <div
              className="lds-spinner"
              onClick={[Function]}
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </Loading>
        </span>
      </Img>
    </Memo(Thumbnail)>
  </div>
</Memo(GalleryComponent)>
`;

exports[`Component GalleryComponent should render a spinner the time to get the answer from the back end. 1`] = `
<Memo(GalleryComponent)
  addImage={[MockFunction]}
  images={Array []}
  isLoading={true}
  onClickThumbnail={[MockFunction]}
  parentUiId="equipment_01"
>
  <div
    className="flex-row top-padding-4px"
  >
    <Label
      className="font-weight-bold"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <label
        className="font-weight-bold"
      >
        <FormattedMessage
          defaultMessage="Gallery"
          description="Gallery title"
          id="gallerytitle"
          values={Object {}}
        >
          Gallery
        </FormattedMessage>
      </label>
    </Label>
    <div>
      <Memo(AddImageFileButton)
        addImage={[MockFunction]}
        className="float-right"
        parentUiId="equipment_01"
      >
        <Button
          aria-label="Edit"
          className="float-right"
          color="light"
          onClick={[Function]}
          tag="button"
        >
          <button
            aria-label="Edit"
            className="float-right btn btn-light"
            onClick={[Function]}
            type="button"
          >
            <span
              className="fa-layers fa-fw"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f03e",
                      "M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z",
                    ],
                    "iconName": "image",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="lg"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-image fa-w-16 fa-lg "
                  data-icon="image"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f067",
                      "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                    ],
                    "iconName": "plus",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="xs"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform="down-13 left-16"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 fa-xs "
                  data-icon="plus"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "transformOrigin": "-0.5625em 1.3125em",
                    }
                  }
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    style={Object {}}
                    transform="translate(224 256)"
                  >
                    <g
                      style={Object {}}
                      transform="translate(-512, 416)  scale(1, 1)  rotate(0 0 0)"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                        style={Object {}}
                        transform="translate(-224 -256)"
                      />
                    </g>
                  </g>
                </svg>
              </FontAwesomeIcon>
            </span>
          </button>
        </Button>
      </Memo(AddImageFileButton)>
    </div>
  </div>
  <div
    className="p-1 border border-secondary rounded shadow gallery"
  >
    <Loading>
      <div
        className="lds-spinner"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </Loading>
  </div>
</Memo(GalleryComponent)>
`;
